test/unit/argument_parser/format_parse_test.cpp

```diff
@@ -1028,6 +956,118 @@ TEST(parse_test, is_option_set)
     EXPECT_THROW(parser.is_option_set('\0'), seqan3::design_error);
 }
 
+namespace foo
+{
+enum class bar
+{
+    one,
+    two,
+    three
+};
+
+auto enumeration_names(bar)
+{
+    return std::unordered_map<std::string_view, bar>{{"one", bar::one}, {"two", bar::two}, {"three", bar::three}};
+}
+} // namespace foo
+
+namespace Other
+{
+enum class bar
+{
+    one,
+    two
+};
+} // namespace Other
+
+namespace seqan3::custom
+{
+template <>
+struct argument_parsing<Other::bar>
+{
+    static inline std::unordered_map<std::string_view, Other::bar> const enumeration_names
+    {
+        {"one", Other::bar::one}, {"two", Other::bar::two}
+    };
+};
+} // namespace seqan3::custom
+
+TEST(parse_type_test, parse_success_enum_option)
+{
+    {
+        foo::bar option_value{};
+
+        const char * argv[] = {"./argument_parser_test", "-e", "two"};
+        seqan3::argument_parser parser{"test_parser", 3, argv, seqan3::update_notifications::off};
+        parser.add_option(option_value, 'e', "enum-option", "this is an enum option.");
+
+        EXPECT_NO_THROW(parser.parse());
+        EXPECT_TRUE(option_value == foo::bar::two);
+    }
+
+    {
+        Other::bar option_value{};
+
+        const char * argv[] = {"./argument_parser_test", "-e", "two"};
+        seqan3::argument_parser parser{"test_parser", 3, argv, seqan3::update_notifications::off};
+        parser.add_option(option_value, 'e', "enum-option", "this is an enum option.");
+
+        EXPECT_NO_THROW(parser.parse());
+        EXPECT_TRUE(option_value == Other::bar::two);
+    }
+}
+
+TEST(parse_type_test, parse_error_enum_option)
+{
+    foo::bar option_value{};
+
+    const char * argv[] = {"./argument_parser_test", "-e", "four"};
+    seqan3::argument_parser parser{"test_parser", 3, argv, seqan3::update_notifications::off};
+    parser.add_option(option_value, 'e', "enum-option", "this is an enum option.");
+
+    EXPECT_THROW(parser.parse(), seqan3::user_input_error);
+}
+
+// https://github.com/seqan/seqan3/issues/2464
+TEST(parse_test, issue2464)
+{
+    using option_t = foo::bar;
+
+    seqan3::value_list_validator enum_validator{(seqan3::enumeration_names<option_t> | std::views::values)};
+    option_t option_value{};
+    std::vector<option_t> option_values{};
```

Comment:
Can we move these into the `{}` test cases?

Reply:
Yes, I will, but isn't that code duplication?